src/rgw/rgw_rest.cc

```diff
@@ -655,7 +655,9 @@ void end_header(struct req_state* s, RGWOp* op, const char *content_type,
       s->formatter->close_section();
     }
     s->formatter->output_footer();
-    dump_content_length(s, s->formatter->get_len());
+    if (s->obj_size) {
```

Comment:
This seems to effect all error response and some require content-length. is it a problem to keep the content-length?